### PR TITLE
impl TryFrom<Payload> for DelegatedAddress

### PR DESCRIPTION
**Pull request by mriise**

### CODE REVIEW

File: shared/src/address/payload.rs [hunk 1]
```diff
@@ -151,6 +151,17 @@ impl From<&Payload> for Protocol {
     }
 }
 
+impl TryFrom<Payload> for DelegatedAddress {
+    type Error = ();
+
+    fn try_from(value: Payload) -> Result<Self, Self::Error> {
+        match value {
+            Payload::Delegated(d) => Ok(d),
+            _ => Err(()),
```
Comment (Stebalien) [inline shared/src/address/payload.rs:160]: We should probably do what we do for ID addresses and introduce a new `Error::NonDelegatedAddress` error. It's not great, but it's better than unit.

File: shared/src/address/payload.rs [hunk 2]
```diff
@@ -151,6 +151,17 @@ impl From<&Payload> for Protocol {
     }
 }
 
+impl TryFrom<Payload> for DelegatedAddress {
```
Comment (Stebalien) [inline shared/src/address/payload.rs:154]: Also implement by reference (not sure if necessary... but idiomatic).

**Review by Stebalien — COMMENTED** [review]
nits but otherwise lgtm.

File: shared/src/address/payload.rs (hunk 2 above)
Comment (mriise) [reply shared/src/address/payload.rs:154]: not really less or more idiomatic IMO, its just whether or not we are going to reuse payload after this call.

**Review by Stebalien — APPROVED** [review]

**Pull request merged**